### PR TITLE
Port tests to python

### PR DESCRIPTION
**Pull request by 1000101**
###### Motivation for this change
#72828

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

**Comment by mmahut** [comment]
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

**Comment by blitz** [comment]
This looks like a pretty straight forward and mechanical conversion. I spotted no issues.
@tfc Ping.

**Review by tfc — APPROVED** [review]

File: nixos/tests/roundcube.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ import ./make-test-python.nix ({ pkgs, ...} : {
   };
 
   testScript = ''
-    roundcube.start
+    roundcube.start()
```
Comment (tfc) [inline nixos/tests/roundcube.nix]: oh good catch... :+1:

**Comment by 1000101** [comment]
Should I drop the tests that are failing on arm or is this good to go?

**Comment by flokli** [comment]
@1000101 did these fail on arm previously, too?

Can you rebase on latest master?

**Comment by 1000101** [comment]
> @1000101 did these fail on arm previously, too?
> 
> Can you rebase on latest master?

Jackett has been failing before:
jackett: https://github.com/NixOS/nixpkgs/pull/72529

Not sure about the hibernate though.

Rebased!

**Comment by mmahut** [comment]
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

**Comment by flokli** [comment]
This is a bit hard to merge, can this be broken down to individual PRs?

**Comment by flokli** [comment]
Ok, it was mostly the roundcube fix already fixed in e292be76fc79dfe1ff1e2680a00c33864caf6a9e. Dropped that commit and rebased on master.

**Comment by flokli** [comment]
Built and succeeded running all updated tests locally (x86_64), merging.

**Comment by 1000101** [comment]
> Ok, it was mostly the roundcube fix already fixed in [e292be7](https://github.com/NixOS/nixpkgs/commit/e292be76fc79dfe1ff1e2680a00c33864caf6a9e). Dropped that commit and rebased on master.

Thanks a million!!!

**Pull request merged**